### ISSUE TITLE
Hybrid development is not working when using ``cds deploy --to hana``

### ISSUE THREAD

**Issue by wattnu** [sapmentors/cap-community#67]
I've found that credentials are not read from ``default-env-json`` if the environment variable NODE_ENV is set to "production". **This prevents from using ``"kind": "sql"`` for a required db in hybrid scenarios where the CAP server is running locally agains a HANA DB on CF.**  ``"kind": "sql"`` is automatically inserted into my package.json when I'm using ``cds deploy --to hana``, so that people are automatically lead into this issue.

Other SAP components, like the Approuter, behave differently, when interpreting ``default-env.json``. An environment variable is taken from this file, if this variable is not set in the real environment. That's why this file is called **default**-env.json, and not just env.json.

As a solution, I would suggest to either consider default-env.json appropriately, or not to use the ``"kind": "sql"`` trick of ``cds deploy --to hana``.

**Some details:**

In my scenario, I've deployed to HANA on CF with ``cd deploy --to hana``. I'm starting the Node.js CAP server and a Fiori UI locally. My package.json service requirement is
```json
    "requires": {
      "db": {
        "kind": "sql"
      }
    }
```
When setting NODE_ENV="" (or "development") and running ``cds watch`` correctly leads to a deployment to SQLite and also my Fiori UI works fine against the CAP service on localhost:4004.

When setting NODE_ENV="production", I was expecting that the kind ``sql`` now resolves to ``hana`` and that the HANA credentials are read from my ``default-env.json``. This is not the case. The log output of ``cds watch`` is showing ``[cds] - connect to hana db {}``. The empty curly braces show me that no credentials are read.  This can be confirmed using the ``cds env ls requires`` command, which shows:
```
requires.db.kind = sql
requires.db.use = hana
requires.hana.use = hana
requires.sql.use = hana
requires.sqlite.credentials.database = sqlite.db
requires.sqlite.use = sqlite
```
When requesting data, the following error occurs (necessarily):
```
[2020-05-04T08:51:26.176Z | ERROR | 1502486]: Cannot read property 'schema' of undefined
[INTERNAL ERROR] TypeError: Cannot read property 'schema' of undefined
    at C:\SAPDevelop\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-hana\client\Client.js:93:31
    at new Promise (<anonymous>)
    at Client.connect (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-hana\client\Client.js:66:12)
    at SQLDatabase._testConnection (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\connect\Service.js:314:18)
    at SQLDatabase._getAndCheckCredentials (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\connect\Service.js:296:17)
    at SQLDatabase._createPool (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\connect\Service.js:243:24)
    at SQLDatabase.acquire (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\connect\Service.js:149:49)
    at SQLDatabase.run (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\connect\Service.js:386:31)
    at OdataContext.context.run (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\services\Service.js:946:30)
    at Object.handler (C:\\Local\prod-cat\node_modules\@sap\cds-runtime\lib\cds-services\services\handlers\onRead.js:45:18)
Please report this error.
```

**Setting ``"kind": "hana"`` and NODE_ENV="development" resolves this issue, but as said above, ``cds deploy --to hana`` is leading to a different setup.**

Versions:
```
>> cds -v
@sap/cds: 3.34.1
@sap/cds-compiler: 1.26.2
@sap/cds-dk: 1.8.0
@sap/cds-foss: 1.2.0
@sap/cds-reflect: 2.11.0
@sap/cds-runtime: 1.1.1
@sap/hana-client: 2.4.196
Node.js: v12.16.3
home: C:\Local\prod-cat\node_modules\@sap\cds
```

**Comment by wattnu** [comment]
Just to add to this ...

Assume, I'm starting with the following configuration:
```json
"requires": {
      "db": {
        "kind": "hana"
      }
    }
```
Running ``cds deploy`` leads to deploying to HANA. After that, my config looks like this:
```json
"requires": {
      "db": {
        "kind": "sql"
      }
    }
```
Running ``cds deploy`` again, now leads to deploying to local SQLite.

IMO, this behavior is really confusing and does not make sense.

Regards

**Comment by chgeo** [comment]
Hi Klaus,

sorry for the late reply, but yes, these seem all valid points to me.  We will try to address them, i.e.
1. Read in `default-env.json` independent from NODE_ENV
2. In `cds deploy -2 hana`, do not set `kind:sql` if `kind:hana` is set already.
3. Check w/ Node.js runtime to improve the error message when HANA credentials are empty.

@joergmann: can you take the 2. issue?

**Comment by chgeo** [comment]
These issues have been fixed in latest sap/cds 3.34.2
